src/plugins/part.c

```diff
@@ -702,74 +702,152 @@ BDPartSpec* bd_part_get_best_free_region (const gchar *disk, BDPartType type, gu
     return ret;
 }
 
+static PedConstraint* _prepare_alignment_constraint (PedDevice *dev, PedDisk *disk, BDPartAlign align, gint *orig_flag_state) {
+    if (align == BD_PART_ALIGN_OPTIMAL) {
+        /* cylinder alignment does really weird things when turned on, let's not
+           deal with it in 21st century (the flag is reset back in the end) */
+        if (ped_disk_is_flag_available (disk, PED_DISK_CYLINDER_ALIGNMENT)) {
+            *orig_flag_state = ped_disk_get_flag (disk, PED_DISK_CYLINDER_ALIGNMENT);
+            ped_disk_set_flag (disk, PED_DISK_CYLINDER_ALIGNMENT, 0);
+        }
+        return ped_device_get_optimal_aligned_constraint (dev);
+    } else if (align == BD_PART_ALIGN_MINIMAL)
+        return ped_device_get_minimal_aligned_constraint (dev);
+    else
+        return NULL;
+}
+
+static void _finish_alignment_constraint (PedDisk *disk, gint orig_flag_state) {
+    if (ped_disk_is_flag_available (disk, PED_DISK_CYLINDER_ALIGNMENT)) {
+        ped_disk_set_flag (disk, PED_DISK_CYLINDER_ALIGNMENT, orig_flag_state);
+    }
+}
+
+static gboolean _resize_part (PedPartition *part, PedDevice *dev, PedDisk *disk, guint64 size, BDPartAlign align, GError **error) {
```

Comment:
Please remove the ``_`` from the start of the functions' names. They are static, there's no point in marking them as internal with any other mechanism.